src/background/background.ts

```diff
@@ -15,33 +11,54 @@ chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
     if (message.command === "startRecordingBytesTransferred") {
         IBytesRepository.instance.clearBytesTransferred();
 
-        chrome.webRequest.onBeforeRequest.addListener(
-            catchPostRequestBodySize,
-            { urls: ["<all_urls>"], tabId },
-            ["requestBody"]
-        );
-        chrome.webRequest.onBeforeSendHeaders.addListener(
-            catchRequestHeaderSize,
-            { urls: ["<all_urls>"], tabId },
-            ["requestHeaders"]
-        );
-        chrome.webRequest.onCompleted.addListener(
-            catchResponseSize,
-            { urls: ["<all_urls>"], tabId },
-            ["responseHeaders"]
-        );
+        chrome.debugger.attach({ tabId: tabId }, "1.2", function () {
+            chrome.debugger.sendCommand(
+                { tabId: tabId },
+                "Network.enable",
+                {},
+                () => {
+                    if (chrome.runtime.lastError) {
+                        console.error(chrome.runtime.lastError);
+                    }
+                }
+            );
+        });
+
         sendResponse(true);
     }
 
     if (message.command === "stopRecordingBytesTransferred") {
-        chrome.webRequest.onCompleted.removeListener(catchResponseSize);
-        chrome.webRequest.onBeforeRequest.removeListener(
-            catchPostRequestBodySize
-        );
-        chrome.webRequest.onBeforeSendHeaders.removeListener(
-            catchRequestHeaderSize
-        );
+        try {
+            await chrome.debugger.detach({ tabId: tabId });
+        } catch (e: unknown) {
```

Comment:
When is `e` not an Error?

Reply:
I think it is always going to be an error -
It's typed as unknown because I don't think [typing in catch clauses are allowed yet](https://github.com/microsoft/TypeScript/issues/42596)

Reply:
In that case, I'd probably remove the if condition and just type cast it as an Error